src/net.cpp

```diff
@@ -408,7 +397,8 @@ CNode* CConnman::FindNode(const CService& addr)
 
 bool CConnman::AlreadyConnectedToAddress(const CAddress& addr)
 {
-    return FindNode(static_cast<CNetAddr>(addr)) || FindNode(addr.ToStringAddrPort());
+    const CService service{MaybeFlipIPv6toCJDNS(addr)};
+    return FindNode(service) || FindNode(service.ToStringAddrPort());
```

Comment:
Before this it would compare just the address and after this change it would compare the address and the port. I think that is undesirable because for inbound connections we see the peer as e.g. `1.2.3.4:somerandomport` and when we try to connect to `1.2.3.4:8333` we should assume we are already connected to that peer.